### PR TITLE
Add release notes for 7.17.0

### PR DESCRIPTION
**Pull request by albertzaharovits**
https://elasticsearch_82858.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.17/release-notes-7.17.0.html

### CODE REVIEW

File: docs/reference/release-notes/7.17.asciidoc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+[[enhancement-7.17.0]]
+[float]
+=== Enhancements
+
+Audit::
+* Emit `trace.id` into audit logs {es-pull}82849[#82849] (issue: {es-issue}74210[#74210])
+
+Authorization::
+* Allow read template with cluster monitor privilege {es-pull}82046[#82046] (issue: {es-issue}78832[#78832])
+* Make authz perf log msg configurable and disabled by default {es-pull}82648[#82648] (issue: {es-issue}75439[#75439])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:30]:
```suggestion
* Make authorization performance log message configurable and disabled by default {es-pull}82648[#82648] (issue: {es-issue}75439[#75439])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+* [Security Solution] Add kibana system permissions for Endpoint action indices {es-pull}81953[#81953]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:31]:
```suggestion
* Add {kib} system permissions for Endpoint action indices {es-pull}81953[#81953]
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+ILM+SLM::
+* Migrate legacy/v2/component templates away from custom attributes routing {es-pull}82472[#82472] (issue: {es-issue}82170[#82170])
+* Migrate to data tiers API dry run on any ILM status {es-pull}82226[#82226] (issue: {es-issue}82169[#82169])
+
+Infra/Core::
+* Allow scaling executors to reject tasks after shutdown {es-pull}81856[#81856] (issues: {es-issue}77017[#77017], {es-issue}77178[#77178])
+* Prevent 6.8 indices upgrade to 8.0 {es-pull}82689[#82689] (issue: {es-issue}81326[#81326])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:39]:
```suggestion
* Prevent direct upgrade of indices from 6.8 to 8.0 {es-pull}82689[#82689] (issue: {es-issue}81326[#81326])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+Infra/Logging::
+* Adjust ILM policy for deprecation logs {es-pull}82833[#82833]
+* Do no use x-opaque-id for deduplicating elastic originating requests {es-pull}82855[#82855] (issue: {es-issue}82810[#82810])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:43]:
```suggestion
* Do no use `x-opaque-id` for deduplicating Elastic originating requests {es-pull}82855[#82855] (issue: {es-issue}82810[#82810])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+Infra/Settings::
+* Implement Setting Deduplication via String Interning {es-pull}80493[#80493] (issues: {es-issue}77466[#77466], {es-issue}78892[#78892])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:46]:
```suggestion
* Implement setting deduplication via string interning {es-pull}80493[#80493] (issues: {es-issue}77466[#77466], {es-issue}78892[#78892])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+Java Low Level REST Client::
+* LLRC: expose http client and allow overriding meta header {es-pull}81955[#81955]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:49]:
```suggestion
* Expose HTTP client and allow overriding meta header {es-pull}81955[#81955]
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+License::
+* Add DEBUG log when DLS/FLS usage is detected {es-pull}82182[#82182] (issue: {es-issue}79152[#79152])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:52]:
```suggestion
* Add `DEBUG` log when document level security and field level security usage is detected {es-pull}82182[#82182] (issue: {es-issue}79152[#79152])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+Machine Learning::
+* Move datafeed stats action off of master node {es-pull}82271[#82271]
+
+Watcher::
+* Preventing watcher from starting up if its templates are missing {es-pull}82395[#82395]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:58]:
```suggestion
* Prevent watcher from starting if its templates are missing {es-pull}82395[#82395]
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+
+
+[[bug-7.17.0]]
+[float]
+=== Bug fixes
+
+Aggregations::
+* Fix bucket keys format for range aggregations on float field {es-pull}81801[#81801] (issue: {es-issue}81749[#81749])
+* Fix cardinality agg in async search {es-pull}82108[#82108]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:68]:
```suggestion
* Fix cardinality aggregation in asynchronous search {es-pull}82108[#82108]
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+* Fix: missing fields in range aggregation response for date fields {es-pull}82732[#82732] (issue: {es-issue}82688[#82688])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:69]:
```suggestion
* Fix missing fields in range aggregation response for date fields {es-pull}82732[#82732] (issue: {es-issue}82688[#82688])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+Authorization::
+* Fix FLS for frozen tier {es-pull}82521[#82521] (issues: {es-issue}78988[#78988], {es-issue}82044[#82044])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:72]:
```suggestion
* Fix field level security for frozen tier {es-pull}82521[#82521] (issues: {es-issue}78988[#78988], {es-issue}82044[#82044])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+Data streams::
+* Change `DataStream` class to have a single public constructor {es-pull}82464[#82464]
+
+Geo::
+* Handle degenerated rectangles in vector tiles {es-pull}82404[#82404] (issue: {es-issue}81891[#81891])
+* `GeoPolygonDecomposer` might fail due to numerical errors when calculating intersection with the dateline {es-pull}82953[#82953] (issue: {es-issue}82840[#82840])
+
+ILM+SLM::
+* Data tiers: migrate the coldest node attribute {es-pull}81940[#81940] (issue: {es-issue}81633[#81633])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:85]:
```suggestion
* Migrate the coldest node attribute for data tier routing {es-pull}81940[#81940] (issue: {es-issue}81633[#81633])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+* ILM step retry safe refresh of the cached phase {es-pull}82613[#82613] (issue: {es-issue}81921[#81921])
+* ILM: fix allocate action to allow only `total_shards_per_node` {es-pull}81944[#81944] (issue: {es-issue}81943[#81943])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:87]:
```suggestion
* Fix allocate action to allow only `total_shards_per_node` {es-pull}81944[#81944] (issue: {es-issue}81943[#81943])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+Data streams::
+* Change `DataStream` class to have a single public constructor {es-pull}82464[#82464]
+
+Geo::
+* Handle degenerated rectangles in vector tiles {es-pull}82404[#82404] (issue: {es-issue}81891[#81891])
+* `GeoPolygonDecomposer` might fail due to numerical errors when calculating intersection with the dateline {es-pull}82953[#82953] (issue: {es-issue}82840[#82840])
+
+ILM+SLM::
+* Data tiers: migrate the coldest node attribute {es-pull}81940[#81940] (issue: {es-issue}81633[#81633])
+* ILM step retry safe refresh of the cached phase {es-pull}82613[#82613] (issue: {es-issue}81921[#81921])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:86]:
```suggestion
* Retry ILM step to safely refresh the cached phase {es-pull}82613[#82613] (issue: {es-issue}81921[#81921])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+* Fix annotations index maintenance after reindexing {es-pull}82304[#82304] (issue: {es-issue}82250[#82250])
+* Improve cleanup for model snapshot upgrades {es-pull}81831[#81831] (issue: {es-issue}81578[#81578])
+* Make delete intervening results more selective {es-pull}82437[#82437]
+* Skip time to next interval with data for datafeeds with aggs {es-pull}82488[#82488] (issue: {es-issue}82406[#82406])
+* Update running process when global calendar changes {es-pull}83044[#83044]
+
+Monitoring::
+* Always attempt upgrade monitoring templates {es-pull}82713[#82713] (issue: {es-issue}82453[#82453])
+
+Other::
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:115]:
```suggestion
Settings::
```

Comment (lockewritesdocs) [reply docs/reference/release-notes/7.17.asciidoc:115]: After renaming this section to **Settings**, I'd move it after line 121 to keep the alphabetic ordering.

File: docs/reference/release-notes/7.17.asciidoc [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+* Checking both node and cluster settings in `NodeDeprecationChecks` {es-pull}82487[#82487] (issue: {es-issue}82484[#82484])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:116]:
```suggestion
* Check both node and cluster settings in `NodeDeprecationChecks` {es-pull}82487[#82487] (issue: {es-issue}82484[#82484])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+* Getting `deprecation.skip_deprecated_settings` to work with dynamic settings {es-pull}81836[#81836]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:117]:
```suggestion
* Change `deprecation.skip_deprecated_settings` to work with dynamic settings {es-pull}81836[#81836]
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+* Ignoring dynamic settings specified by `deprecation.skip_deprecated_settings` in node deprecation checks {es-pull}82883[#82883] (issue: {es-issue}82889[#82889])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:118]:
```suggestion
* Ignore dynamic settings specified by `deprecation.skip_deprecated_settings` in node deprecation checks {es-pull}82883[#82883] (issue: {es-issue}82889[#82889])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+
+Search::
+* Fix bug where field is not returned if it has the same prefix as a nested field {es-pull}82922[#82922] (issue: {es-issue}82905[#82905])
+
+Snapshot/Restore::
+* Always fail snapshot deletion listeners on master fail over {es-pull}82361[#82361] (issue: {es-issue}81596[#81596])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:124]:
```suggestion
* Always fail snapshot deletion listeners on master failover {es-pull}82361[#82361] (issue: {es-issue}81596[#81596])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+* Fix Potential Repository Corruption during Master Failover {es-pull}82912[#82912] (issue: {es-issue}82911[#82911])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:125]:
```suggestion
* Fix potential repository corruption during master failover {es-pull}82912[#82912] (issue: {es-issue}82911[#82911])
```

File: docs/reference/release-notes/7.17.asciidoc [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+* Remove Requirement for Key Setting on Azure Client Settings {es-pull}82030[#82030]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/7.17.asciidoc:126]:
```suggestion
* Remove requirement for key setting on Azure client settings {es-pull}82030[#82030]
```

**Review by lockewritesdocs — COMMENTED** [review]
Suggested some grammatical and editorial changes, but LGTM otherwise.

**Comment by albertzaharovits** [comment]
Thanks for the speedy review @lockewritesdocs . I have acknowledged all your points.

**Review by lockewritesdocs — APPROVED** [review]
👍 thanks @albertzaharovits!

File: docs/reference/release-notes/7.17.asciidoc [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+* Fix missing fields in range aggregation response for date fields {es-pull}82732[#82732] (issue: {es-issue}82688[#82688])
+
+Authorization::
+* Fix field level security for frozen tier {es-pull}82521[#82521] (issues: {es-issue}78988[#78988], {es-issue}82044[#82044])
+
+CRUD::
+* Fix potential listener leak in `TransportBulkAction` {es-pull}81894[#81894]
+
+Data streams::
+* Change `DataStream` class to have a single public constructor {es-pull}82464[#82464]
```
Comment (droberts195) [inline docs/reference/release-notes/7.17.asciidoc:78]: Not sure this should be in the release notes.  Why would end users care?

**Pull request merged**